functions/Get-DbaJobOutputFile.ps1

```diff
@@ -0,0 +1,201 @@
+         $table.columns.add($col1)
+         $table.columns.add($col2)
+         $table.columns.add($col3)
+         $table.columns.add($col4)
+        function Add-Row 
+        {
+        # Create a new Row
+            $row = $table.NewRow() 
+            # Add values to new row
+            $row.Server = $Server.ComputerNamePhysicalNetBIOS
+            $row.Instance = $Server.Instancename
+            $row.Job = $Job.Name
+            $row.JobStep = $step.Name
+            $row.FileName = $FileName
+          
+            #Add new row to table
+            $table.Rows.Add($row)
+        }
+         
+    }
+    PROCESS
+    {
+        foreach ($instance in $sqlserver)
+        {
+            $Server = Connect-SqlServer -SqlServer $instance  -SqlCredential $SqlCredential
+            $jobs = $Server.JobServer.Jobs
+            if ($JobName)
+            {
+                $Job = $server.JobServer.Jobs[$JobName]
+                $Servername = $Server.ComputerNamePhysicalNetBIOS
+                foreach($Step in $Job.JobSteps)
+                {
+                $fileName = $Step.OutputFileName
+                if($fileName -eq '')
+                {
+                    Add-Row 
+                }
+                else
+                {         
+                    if($FileName.StartsWith('\\') -eq $false)
+            {
+                $fileName = '\\' + $Servername + '\' + $Filename.Replace(':','$')
+                Add-Row 
+            }
+            else
+            {
+              Add-Row 
+            }
+            if($OpenFile)
+            {
+                if(Test-Path $fileName)
+                {
+                    If ($Pscmdlet.ShouldProcess("$FileName", "Opening File $FileName"))
+				    {
+                        Invoke-Item $fileName
+                    }
+                }
+                else
+                {
+                    Write-Output 'No File to open'
+                }
+            }
+        }
+            }
+            }
+            else
+            {
+                foreach($Job in $Jobs)
+                {
+            $Servername = $Server.ComputerNamePhysicalNetBIOS
+            foreach($Step in $Job.JobSteps)
+            {
+                $fileName = $Step.OutputFileName
+                if($fileName -eq '')
+                {
+                    Add-Row 
+                }
+                else
+                {        
+                    if($FileName.StartsWith('\\') -eq $false)
+                    {
+                $fileName = '\\' + $Servername + '\' + $Filename.Replace(':','$')
+                Add-Row 
+            }
+                    else
+                    {
+                Add-Row 
+            }
+                    if($OpenFile)
+                    {
+                if(Test-Path $fileName)
+                {
+                    If ($Pscmdlet.ShouldProcess("$FileName", "Opening File $FileName"))
+				    {
+                        Invoke-Item $fileName
```

Comment:
Sorta like I have to remove Out-GridView from Show-SqlSpWhois, I think this should be left to the user. You can return LocalFileName and RemoteFileName which is what I do in Find-DbaOrphanedFile so ppl can pipe that to invoke or remove or whatever.